src/platform/forms-system/src/js/components/Telephone.jsx

```diff
@@ -0,0 +1,145 @@
+  const len = number.length - 1;
+  // return rounded number as a grouped value, e.g. `800` or `1000`
+  // but split non-round numbers, e.g. `827` => `8 2 7`
+  const roundNumber = parseInt(number.slice(-len), 10) === 0;
+  return roundNumber ? number : number.split('').join(' ');
+};
+
+// Combine phone number blocks within the label separated by periods
+// "800-555-1212" => "800. 5 5 5. 1 2 1 2"
+const formatTelLabel = number =>
+  number
+    .split(/[^\d]+/)
+    .filter(n => n)
+    .map(formatBlock)
+    .join('. ');
+
+/**
+ * Produce a standardized accessible telephone link
+ * @param {string} tel - Telephone number (with or without leading "1")
+ * @param {string} use - Known phone number label, e.g. "gibill"
+ * @param {string} className - Additional class name to add
+ * @param {string} pattern - Formatted pattern using "#" as placeholders
+ * @param {string|function} label - Custom aria-label string/function
+ * @param {string|function} text - Custom link text string/function
+ * @param {function} onClick - Custom onClick function
+ */
+export default function Telephone({
+  // phone number (length _must_ match the pattern; leading "1" is removed)
+  tel = '',
+  use = '', // quick labels for known numbers
+  className = '', // additional css class to add
+  pattern = '', // output format; defaults to patterns.default value
+  label = '', // custom aria-label
+  text = '', // custom text formatting
+  onClick = recordClick,
+}) {
+  // "use" is for known numbers
+  const useString = formatLabel(use.toString());
+  const knownNumber = useString
+    ? Object.keys(knownNumbers).find(number =>
+        knownNumbers[number].find(lbl =>
+          useString.startsWith(lbl.substring(0, 4)),
+        ),
+      )
+    : tel.toString();
+
+  if (useString && !knownNumber) {
+    const known = Object.entries(knownNumbers).map(
+      ([number, labels]) =>
+        ` ${formatTelText(
+          number,
+          patterns[useString] || patterns.default,
+        )}: ${labels.join(', ')}`,
+    );
+    throw new Error(
+      `Telephone: "${useString}" label is not recognized; Predefined list:\n` +
+        `${known.join('\n')}`,
+    );
+  }
+
+  // format href: "###-### ####" => "##########"
+  const number = cleanHref(knownNumber);
+
+  const usePattern = pattern || patterns[useString] || patterns.default;
+  const patternLength = usePattern.match(/#/g).length;
+  const formattedTel = formatTelText(number, usePattern);
+
+  if (!useString && (!number || number.length !== patternLength)) {
+    throw new Error(
+      `Telephone: "${number}" does not match the pattern (${usePattern})`,
+    );
+  }
+
+  const href = number.length === 10 ? `+1${number}` : number;
+
+  const ariaLabel =
+    typeof label === 'function'
+      ? label(formattedTel)
+      : label || formatTelLabel(formattedTel);
+
+  const numberText =
+    typeof text === 'function'
+      ? text(number, usePattern)
+      : text || formattedTel;
```

Comment:
A `render` prop that expects a `ariaLabel` and a `text` to be returned might feel a little more natural as a component API and be easier to work with in Telephone -

```jsx
<Telephone render={number => ({ ariaLabel: 'Label', text: 'Text' })}/>
```

Reply:
Hmm, maybe instead of `render`, we _could_ include a child node (contrary to what I said before) that takes the place of the generated text. And also include an `aria-label` prop that overrides that generated content. What do you think?

```html
<Telephone contact={CONTACTS.CRISIS_LINE} ariaLabel={"1. 800. 2 7 3. 8 2 5 5.">
   (800) 273-8255
</Telephone>
```

Reply:
I think it makes perfect sense to have that as sort of a catch-all to make the component flexible for use cases that we don't know of yet! If a `contact` is passed, I would hope that they wouldn't need to override the label/child node because they'll have to hardcode the phone number again, but that could be a necessary risk

Reply:
Implemented! Thanks for the awesome idea!